node/runtime/src/impls.rs

```diff
@@ -44,27 +44,38 @@ impl Convert<u128, Balance> for CurrencyToVoteHandler {
 ///
 /// This assumes that weight is a numeric value in the u32 range.
 ///
+/// Given `TARGET_BLOCK_FULLNESS = 1/2`, a block saturation greater than 1/2 will cause the system
+/// fees to slightly grow and the opposite for block saturations less than 1/2.
+///
 /// Formula:
-///   diff = (ideal_weight - current_block_weight)
+///   diff = (target_weight - current_block_weight)
 ///   v = 0.00004
 ///   next_weight = weight * (1 + (v . diff) + (v . diff)^2 / 2)
 ///
 /// https://research.web3.foundation/en/latest/polkadot/Token%20Economics/#relay-chain-transaction-fees
 pub struct WeightMultiplierUpdateHandler;
+
+// CRITICAL NOTE: The system module maintains two constants: a _maximum_ block weight and a
+// _ratio_ of it yielding the portion which is accessible to normal transactions (reserving the rest
+// for operational ones). `TARGET_BLOCK_FULLNESS` is entirely independent and the system module is
+// not aware of if, nor should it care about it. This constant simply denotes on which ratio of the
+// _maximum_ block weight we tweak the fees. It does NOT care about the type of the dispatch.
+//
+// For the system to be configured in a sane way, `TARGET_BLOCK_FULLNESS` should always be less than
+// the ratio that `system` module uses to find normal transaction quota.
+/// The block saturation level. Fees will be updates based on this value.
+pub const TARGET_BLOCK_FULLNESS: Perbill = Perbill(1_000_000_000 / 4);
```

Comment:
This is one of the items that we might want to place elsewhere. I think it actually looks okay to be in this crate `node/runtime` but preferably in a separate `consts.rs` file? We will probably have other configurable constants in the future as well.

Reply:
did as explained for now. Moved some constants to a new file.

Reply:
language seems to flip between 1/4 of block is reserved for operation transactions and 3/4 of block is reserved. here, for example `TARGET_BLOCK_FULLNESS` is 25%, indicating 3/4 reserved for operational transactions. however, in the diagram at the top of this page, it says:

>  25% ... Reserved for Operational 

Please consult research team and get this consistent and correct.

Reply:
This is exactly what I addressed in the comment above this line:

```
// CRITICAL NOTE: The system module maintains two constants: a _maximum_ block weight and a
// _ratio_ of it yielding the portion which is accessible to normal transactions (reserving the rest
// for operational ones). `TARGET_BLOCK_FULLNESS` is entirely independent and the system module is
// not aware of if, nor should it care about it. This constant simply denotes on which ratio of the
// _maximum_ block weight we tweak the fees. It does NOT care about the type of the dispatch.
```

To rephrase: They are entirely separate. 

System module knows:
  - Each block can be a __total__ of `X weight` units. 
  - It will keep `25%` of this only for operational transactions and the rest for normal.

Weight Multiplier:
  - Asks the system module for the __total__ weight units per block. 
  - Assumes that if we are `25%` of this limit then we are _relatively congested_ and should slightly increase the weight fees. 

The only adjustment I see is that the Weight multiplier should query the system module for the __available-to-normal-txs__ portion of weights and set its goal for 25% of that. But I still think it makes more sense like this. Weight multiplier should not care about `Operational` (heck, it even doesn't know they exist). Only note will be, as commented in the code: 

```
// For the system to be configured in a sane way, `TARGET_BLOCK_FULLNESS` should always be less than
// the ratio that `system` module uses to find normal transaction quota.
```

Agreeing that the research team should verify the numbers.

Reply:
ok cool